conan/internal/api/detect_api.py

```diff
@@ -568,6 +569,9 @@ def default_compiler_version(compiler, version):
     of the minor or patch digits, that do not affect binary compatibility
     """
     output = ConanOutput(scope="detect_api")
+    if not version:
+        raise ConanException(
+            f"No version provided to default_compiler_version for {compiler} compiler")
```

Comment:
We also thought that instead of this, we could add prints to the `None, None, None` cases in the detect functions, but I think that ends up being quite noisy, so this approach feels better